packages/fuels-accounts/src/provider.rs

```diff
@@ -754,4 +749,26 @@ impl DryRunner for Provider {
     fn consensus_parameters(&self) -> &ConsensusParameters {
         self.consensus_parameters()
     }
+
+    async fn maybe_estimate_predicates(
+        &self,
+        tx: &FuelTransaction,
+        latest_chain_executor_version: Option<u32>,
+    ) -> Result<Option<FuelTransaction>> {
+        let latest_chain_executor_version = match latest_chain_executor_version {
+            Some(exec_version) => exec_version,
+            None => {
+                self.chain_info()
+                    .await?
+                    .latest_block
+                    .header
+                    .state_transition_bytecode_version
+            }
+        };
+
+        Ok(
+            (latest_chain_executor_version > LATEST_STATE_TRANSITION_VERSION)
```

Comment:
@xgreenx do we also need to check the consensus values version in case the gas prices change?

Reply:
Do we want to hold off the merge until this is answered?

Reply:
No, we don't need to check consensus parameters for estimation. We may want to check remote version with local version to be sure that we use the latest consensus parameters. If local consensus parameters outdated, then we want to download them one more time.

We can do that because without new state transition function `fuel-core` can't support new gas costs variant -> If the state transition function is the same, but consensus parameters are different, we know how to download them since it is supported version.

Reply:
> We may want to check remote version with local version to be sure that we use the latest consensus parameters. If local consensus parameters outdated, then we want to download them one more time.

I believe we already do this. I need to double-check, though. cc @hal3e

Reply:
Currently we cache the `consensus_parameters` once when we connect to the node. I do not see any update mechanism. We will have to add this.

Reply:
Yeah, we're not updating it, we just check the supported version during the build.

Reply:
Update: We'll implement this update mechanism soon as part of a follow-up work, so we don't block the release here.